tests/test_crossdock.py

```diff
@@ -24,143 +24,126 @@
 if six.PY2:
     from crossdock.server import server
 from tornado.httpclient import HTTPRequest
+import tornado.testing
 from jaeger_client import Tracer, ConstSampler
 from jaeger_client.reporter import InMemoryReporter
 from crossdock.server.endtoend import EndToEndHandler, _determine_host_port, _parse_host_port
 
 tchannel_port = "9999"
-
-
-@pytest.fixture
-def app():
-    """Required by pytest-tornado's http_server fixture"""
-    s = server.Server(int(tchannel_port))
-    s.tchannel.listen()
-    return server.make_app(s, EndToEndHandler())
-
-
-# noinspection PyShadowingNames
-@pytest.yield_fixture
-def tracer():
```

Comment:
why do you need to change this? it's not dependent on pytest-tornado. If you simply call get_tracer() from the unit test, then you're never closing the tracer.

Reply:
I tried it. Using any generators within a test case without marking it as `@tornado.testing.gen_test` causes an error. If I add that, it won't work at all because it is not meant for object methods, only free functions.

Reply:
does `tornado.testing` not work with free functions at all? I have that class extension pattern, makes a lot of testing annotations unusable.

Reply:
I totally agree about that. On the other hand, I saw no way around this given that pytest-tornado hasn't been maintained in a while and pytest has deprecated some key features.